js/GroupedBarChart.js

```diff
@@ -63,263 +120,249 @@ var GroupedBarChart = function(param)
             var sort = data[0].breakdowntype !== "STRING";
```

Comment:
//or maybe somewhere further up. 
//These are the only valid data types.
isValidData = (data[0].breakdowntype === "STRING" || data[0].breakdowntype === "YMD" || data[0].breakdowntype === "YM" || data[0].breakdowntype === "YW");  

isDateData =  isValidData && data[0].breakdowntype !== "STRING";

//I enjoy a boolean variable convention if "isCondition" for readability
return isCondition? x: y;